File: m/rls/rlb/stsam.c
/* file -  stsam - strings same test */
#include "m:\rid\rider.h"
#include "m:\rid\stdef.h"
/* code -  st_sam - string equality test */
int st_sam(
register char *mod ,
register char *tar )
{ while ( *mod++ == *tar) {
    if( *tar++ == 0)return 1;
  } 
  return 0;
} 

File: m/rls/stsam.r
file	stsam - strings same test
include	rid:rider
include	rid:stdef

code	st_sam - string equality test

  func	st_sam 
	mod : * char~			; model string
	tar : * char~			; target string
	()  : int			; fine => strings same
  is	while *mod++ eq *tar		; are same
	   fine if *tar++ eq		; perfect match
	end				;
	fail				;
  end
